source/deployment/sso-saml-okta.rst

```diff
@@ -0,0 +1,101 @@
+
+1. Sign into Okta as an administrator.
+
+2. Go to **Admin Dashboard > Applications > Add Application**.
+
+3. Click **Create New App** and choose ``SAML 2.0`` as the Sign on method.
+
+	.. image:: ../../source/images/okta_1_new_app.PNG
+
+4. Enter **General Settings** for the application, including ``App name`` and ``App logo`` (optional). It is recommended to display the application icon to users, including in the Okta Mobile app.
+
+  If you’d like to use a Mattermost logo for the application, you are free to download one `from our page <http://www.mattermost.org/brand-guidelines/>`_.
+
+	.. image:: ../../source/images/okta_2_general_settings.PNG
+
+5. Enter **SAML Settings**, including:
+ - Single sign on URL: ``https://<your-mattermost-url>/login/sso/saml``
+ - Audience URL: For instance, ``mattermost``
+ - Name ID format: ``unspecified``
+ - Application username: ``Email``
+
+	.. image:: ../../source/images/okta_3_initial_saml_settings.PNG
+
+6. (Optional) Set up encryption for your SAML connection. First, click **Show Advanced Settings**.
+
+	.. image:: ../../source/images/okta_4_initial_saml_settings.PNG
+
+Then, set **Assertion Encryption** as ``Encrypted`` and upload the **Service Provider Public Certificate** you generated in step 2 to the **Encryption Certificate** field.
+
+	.. image:: ../../source/images/okta_5_advanced_saml_settings.PNG
+
+7. Enter attribute statements, which will be used to map attributes between Okta and Mattermost. For more information on which attributes are configurable, see our :ref:`documentation on SAML configuration settings <saml-enterprise>`. Email and username attributes are required. For Mattermost servers running 3.3 and earlier, first name and last name attributes are also required.
+
+	.. image:: ../../source/images/okta_6_attribute_statements.PNG
+
+8. Click **Next**. Then, set Okta support parameters for the application. Recommended settings:
+ - I’m an Okta customer adding an internal app
+ - This is an internal app that we have created
+
+	.. image:: ../../source/images/okta_7_support_configuration.PNG
+
+9. Click **Finish**. On the next screen, click the **Sign On** tab and click **View Setup Instructions**.
+
+	.. image:: ../../source/images/okta_8_view_instructions.PNG
+
+10. Take note of ``Identity Provider Single Sign-On URL`` (also known as ``SAML SSO URL``), and the Identity Provider Issuer, as both will be needed to configure SAML for Mattermost.
+
+Furthermore, you **must download the X.509 Public Certificate file** and save it. You will need to upload it to Mattermost at a later step.
+
+	.. image:: ../../source/images/okta_9_view_instructions.PNG
+
+Configure SAML for Mattermost
+-----------------------------
+
+1. Start Mattermost server and sign into Mattermost as a System Administrator. Go to **System Console > Authentication > SAML**, and enter the following fields:
+ - **SAML SSO URL**: ``Identity Provider Single Sign-On URL`` from Okta, specified earlier.
```

Comment:
I didn't update the docs that weren't part of the original PR, (OneLogin SAML). I only converted from Markdown to reStructuredText, and left style etc for another time.